frontend/src/bundles/common/components/link-preview/link-preview.tsx

```diff
@@ -0,0 +1,101 @@
+import { getValidClassNames } from '../../helpers/helpers.js';
+import { useEffect, useState } from '../../hooks/hooks.js';
+import { Grid, Typography } from '../components.js';
+import { fetchLinkPreview } from './fetch-link-preview.helper.js';
+import { type LinkPreviewData } from './link-preview-data.type.js';
+import styles from './styles.module.scss';
+
+type Properties = { url: string; withHeader?: boolean };
+
+const LinkPreview: React.FC<Properties> = ({ url, withHeader = false }) => {
+    const [previewData, setPreviewData] = useState<LinkPreviewData | null>(
+        null,
+    );
+    const [loading, setLoading] = useState(true);
+
+    useEffect(() => {
+        const fetchData = async (): Promise<void> => {
+            const data = await fetchLinkPreview(url);
+            setPreviewData(data);
+            setLoading(false);
+        };
+
+        void fetchData();
+    }, [url]);
+
+    if (loading) {
+        return (
+            <Grid
+                container
+                className={getValidClassNames(
+                    styles.container,
+                    styles.skeleton,
+                )}
+            >
+                {withHeader && (
+                    <>
+                        <div
+                            className={getValidClassNames(
+                                styles.title,
+                                styles.skeleton,
+                            )}
+                        ></div>
+                        <div
+                            className={getValidClassNames(
+                                styles.description,
+                                styles.skeleton,
+                            )}
+                        ></div>
+                    </>
+                )}
+                <Grid
+                    container
+                    item
+                    className={getValidClassNames(
+                        styles.imgWrapper,
+                        styles.skeleton,
+                    )}
+                >
+                    Loading...
+                </Grid>
+            </Grid>
+        ); // TODO: add loader or skeleton
```

Comment:
why we need this comment?

Reply:
It has a very simple design without animation, etc.
And it can be added in the future, because now I can spend a lot of time preparing design.